### PR TITLE
345: Dependencies

### PR DESCRIPTION
**Pull request by pimotte**
In the end without conditional votes, but I think this is the most consistent formulation considering the other rules.

### CODE REVIEW

**Comment by MrHug** [comment]
This looks like a very clean and consistent attempt. I will look at it again tomorrow, but so far have no objections :)

File: README.md [hunk 1]
```diff
@@ -293,6 +293,28 @@ or an empty string. Conditional votes are a comment representing a vote.
...
+**345** *Dependency Hell*
+
+The owner of a PR may declare one dependency of this PR (henceforth "child") on another
+PR (henceforth "parent"), by including the text
+"This PR is dependent on #<number of parent>" in an unedited post. Any votes above
+this post are voided. A child PR must be of a branch that branches off the
+branch of the parent.
+
+A ":+1:" vote on a child implies a ":+1:" on its parent. This vote shall be read
+as a comment on the parent just before merging for the purposes of other rules.
```
Comment (MrHug) [inline README.md]: Hmm, maybe keep this generic: A vote in favour

After all the ":+1:" format could change ;)

File: README.md [hunk 2]
```diff
@@ -293,6 +293,28 @@ or an empty string. Conditional votes are a comment representing a vote.
 In order to perform an action based on resolution of one of more conditional votes,
 a player must post a comment describing the resolution before performing said action.
 
+**345** *Dependency Hell*
+
+The owner of a PR may declare one dependency of this PR (henceforth "child") on another
+PR (henceforth "parent"), by including the text
+"This PR is dependent on #<number of parent>" in an unedited post. Any votes above
+this post are voided. A child PR must be of a branch that branches off the
+branch of the parent.
```
Comment (MrHug) [inline README.md]: Must the child be a PR on master, or a PR on the parent?

**Comment by MrHug** [comment]
Hmm, looking better already :)

What about the order in which stuff is merged? Because merging the child first would be a violation I guess? So parent first then child? If so, do we need to document this?

**Comment by pimotte** [comment]
The child needs to be branched off the parent. The thought is that pressing "Merge pull request" on the child will automatically merge the parent (and would constitute the "non-seperate" merging specified in this rule) There could be a problem if the parent has diverged and the child hasn't merged that back in. I could explicitly add that, but as far as I'm concerned, the only the current proposal allows is to merge the parent back in to the child.

**Comment by MrHug** [comment]
Hmmm ok, I'm just wondering how GH treats that... As we base our points on merged PRs into master, I wonder if GH will see this as merging 2 PRs when when merge the child, or whether it will merge only the child and leave the parent in some weird as state that leads to a closed PR. In the latter case our scoring system could be tougher to check...?

**Comment by pimotte** [comment]
I'm about 98% sure it will merge the parent PR, in the sense that for github "merged" is not a state. PRs are either open or closed, and a closed PR is merged iff the comparison of that branch to master leads to "nothing to be merged"

**Comment by MrHug** [comment]
Hmm ok, fair enough

**Comment by MrHug** [comment]
:+1:

**Comment by pimotte** [comment]
@jdonkervliet @ManInTheGitHub This PR enables a lot of fun stuff, and both Stefan and I have some ideas that depend on this ;)

**Comment by jdonkervliet** [comment]
:+1:

**Comment by pimotte** [comment]
3/4 votes is merge.

**Pull request merged**